tests/dispatch/tests.py

```diff
@@ -153,6 +196,17 @@ def test_disconnection(self):
         a_signal.disconnect(receiver_3)
         self.assertTestIsClean(a_signal)
 
+    def test_disconnect_weak_deprecated(self):
```

Comment:
Again this is mostly unrelated, but it's mildly inconvenient when deprecated tests are removed because it often creates backport conflicts due to removal of `import warnings` and the tests themselves. I was thinking maybe a policy of putting deprecation tests in a separate file (e.g. `dispatch/test_removedindjango20.py`) could mitigate this a bit while also easing removal of deprecated tests.